src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -512,6 +514,14 @@ private void uploadFleetStatusServiceData(Set<GreengrassService> greengrassServi
                                               OverallStatus overAllStatus,
                                               DeploymentInformation deploymentInformation,
                                               Trigger trigger) {
+        // Only allow component state change update publish if FSS set up is complete
+        // If set up is incomplete, it may cause a deadlock
+        if (!isLaunchMessageSent.get() && !Trigger.NUCLEUS_LAUNCH.equals(trigger)) {
```

Comment:
Use compare and set, not get. Read the return value from compare and set to know what action you should take.

Reply:
lemme open a new pr for this